src/test/java/seedu/address/model/transaction/BuyTransactionTest.java

```diff
@@ -0,0 +1,101 @@
+package seedu.address.model.transaction;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalGoods.APPLE;
+import static seedu.address.testutil.TypicalGoods.BANANA;
+import static seedu.address.testutil.TypicalGoods.CITRUS;
+import static seedu.address.testutil.TypicalSuppliers.ALICE;
+import static seedu.address.testutil.TypicalSuppliers.BENSON;
+import static seedu.address.testutil.TypicalSuppliers.CARL;
+import static seedu.address.testutil.TypicalTransactions.BUY_APPLE_TRANSACTION;
+
+import java.util.UUID;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.offer.Price;
+import seedu.address.testutil.BuyTransactionBuilder;
+
+public class BuyTransactionTest {
+
+    public static final String DEFAULT_ID = UUID.randomUUID().toString();
+    private static final String DEFAULT_PRICE = "26.58";
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () ->
+                new BuyTransaction(null, APPLE, ALICE, new Price(DEFAULT_PRICE)));
+        assertThrows(NullPointerException.class, () ->
+                new BuyTransaction(new TransactionId(DEFAULT_ID), null, BENSON, new Price(DEFAULT_PRICE)));
+        assertThrows(NullPointerException.class, () ->
+                new BuyTransaction(new TransactionId(DEFAULT_ID), BANANA, null, new Price(DEFAULT_PRICE)));
+        assertThrows(NullPointerException.class, () ->
+                new BuyTransaction(new TransactionId(DEFAULT_ID), CITRUS, CARL, null));
+    }
+
+
+    @Test
+    public void isSameBuyTransactionTest() {
+
+        BuyTransaction buyTransaction = new BuyTransactionBuilder(BUY_APPLE_TRANSACTION).build();
+
+        // same object -> returns true
+        assertTrue(buyTransaction.isSameBuyTransaction(buyTransaction));
+
+        // null -> returns false
+        assertFalse(buyTransaction.isSameBuyTransaction(null));
+
+        // different id, same good, same person, same buy price -> returns false
+        BuyTransaction editedBuyTransaction = new BuyTransactionBuilder(buyTransaction)
+                .withId(DEFAULT_ID).build();
+        assertFalse(buyTransaction.isSameBuyTransaction(editedBuyTransaction));
```

Comment:
So, DEFAULT_ID is not the ID of the BUY_APPLE_TRANSACTION?

Reply:
Name may sound a bit misleading

Reply:
all typical transactions' id was manually defined.